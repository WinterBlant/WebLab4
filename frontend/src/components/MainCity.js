import React from 'react';
import '../styles/main.css';

const MainCity = ({ weather }) => (
    !!weather ?
        <div className="currentCity">
            <h1>
                {weather.name}
            </h1>
            <div className="additionalInfo">
                <img
                    className="weatherImage"
                    src={weather.img}
                    alt='Иконка погоды'>
                </img>
                <h1>
                    {weather.temperature}
                </h1>
            </div>
        </div>
        :
        null
);


export default MainCity;