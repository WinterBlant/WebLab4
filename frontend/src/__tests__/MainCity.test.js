import MainCity from '../components/MainCity';
import renderer from 'react-test-renderer';
import React from 'react';

describe('CurrentCityComponent component', () => {
    test('matches the snapshot when the weather object is not null', () => {
        const weather = {
            name: 'Saint Petersburg',
            img: '123',
            temperature: '5'
        };
        const tree = renderer.create(
            <MainCity weather={weather}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when the weather object is null', () => {
        const weather = null;
        const tree = renderer.create(
            <MainCity weather={weather}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});