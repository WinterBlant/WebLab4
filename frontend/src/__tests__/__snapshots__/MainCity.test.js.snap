// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCityComponent component matches the snapshot when the weather object is not null 1`] = `
<div
  className="currentCity"
>
  <h1>
    Saint Petersburg
  </h1>
  <div
    className="additionalInfo"
  >
    <img
      alt="Иконка погоды"
      className="weatherImage"
      src="123"
    />
    <h1>
      5
    </h1>
  </div>
</div>
`;

exports[`CurrentCityComponent component matches the snapshot when the weather object is null 1`] = `null`;
